tensorflow/contrib/kinesis/python/ops/kinesis_op_loader.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Python helper for loading kinesis ops and kernels."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.contrib.util import loader
+from tensorflow.python.platform import resource_loader
+
+_dataset_ops = loader.load_op_library(
+    resource_loader.get_path_to_datafile("../../_dataset_ops.so"))
```

Comment:
Do we build and load an empty object file in the default build configuration where this is not enabled? Is there a way to avoid this?

Reply:
Thanks @mrry. At the moment, the `//tensorflow/contrib/kinesis` build is controlled by `with_aws_support` flag. The `//tensorflow/contrib/kinesis` build is skipped with `with_aws_support=False`. So the object file will not be built in that case.